### PR TITLE
Horiz amb occl performance updates

### PR DESCRIPTION
**Pull request by vastcharade**
Interview test project assigned by Brad Hefta-Gaub in an attempt to improve the performance of the horizon ambient occlusion algorithm. The changes have increased performance by 20fps on a macbook pro, but I'm not confident the algorithm is working properly. It works reasonably well for the default avatar scene, but other scenes have not been tested against.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by vastcharade** [comment]
ok to test

**Comment by ZappoMan** [comment]
Ok to test

**Comment by ZappoMan** [comment]
ok to test

**Comment by hifi-gustavo** [comment]
Build console output for commit: 21dc58eb85fcf795f98fa8e7be7d24ccc1274e10 available at: https://gist.github.com/35d7b5c4559bacdddd1b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5577.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5577.dmg

File: libraries/render-utils/src/ambient_occlusion.slf [hunk 1]
```diff
@@ -30,25 +30,40 @@ uniform float g_scale;
...
 
 const float AOStrength = 1.9;
-const float R = 0.3;
-const float R2 = 0.3*0.3;
-const float NegInvR2 = - 1.0 / (0.3*0.3);
+
+
+// TODO: R (radius) should be exposed as a uniform parameter
+const float R = 0.01;
+const float R2 = 0.01*0.01;
```
Comment (ZappoMan) [inline libraries/render-utils/src/ambient_occlusion.slf]: coding standard: 4.2.1. Conventional operators should be surrounded by a space character

https://docs.highfidelity.com/docs/coding-standard#section-4-2-1-conventional-operators-should-be-surrounded-by-a-space-character-

File: libraries/render-utils/src/ambient_occlusion.slf [hunk 2]
```diff
@@ -30,25 +30,40 @@ uniform float g_scale;
...
+const float NegInvR2 = - 1.0 / (0.01*0.01);
```
Comment (ZappoMan) [inline libraries/render-utils/src/ambient_occlusion.slf]: coding standard: 4.2.1. Conventional operators should be surrounded by a space character

https://docs.highfidelity.com/docs/coding-standard#section-4-2-1-conventional-operators-should-be-surrounded-by-a-space-character-

File: libraries/render-utils/src/ambient_occlusion.slf [hunk 3]
```diff
@@ -56,113 +71,126 @@ const int NumSamples = 4;
...
+    return normalize(texture(normalTexture, uv).xyz - 0.5);
+}
 
-	// Get view space Z
-	return -1.0 / (LinMAD.x * d + LinMAD.y);
+/**
+ * Gets the linearized depth in view space.
+ * d: the depth value [0-1], usually from a depth texture to convert.
+ */
+float ViewSpaceZFromDepth(float d){
```
Comment (ZappoMan) [inline libraries/render-utils/src/ambient_occlusion.slf]: coding standard: 1.2.12. Abbreviations in names should be avoided

https://docs.highfidelity.com/docs/coding-standard#section-1-2-12-abbreviations-in-names-should-be-avoided-

File: libraries/render-utils/src/ambient_occlusion.slf [hunk 4]
```diff
@@ -180,69 +207,73 @@ void ComputeSteps(inout vec2 stepSizeUv, inout float numSteps, float rayRadiusPi
...
+        // Calculate the horizon occlusion of each direction
+        for(float d = 0; d < numDirections; ++d) {
+            float theta = alpha * d;
+
+            // Apply noise to the direction
+            vec2 dir = RotateDirections(vec2(cos(theta), sin(theta)), random.xy);
+            vec2 deltaUV = dir * stepSizeUV;
+
+            // Sample the pixels along the direction
+            ao += HorizonOcclusion(	deltaUV,
```
Comment (ZappoMan) [inline libraries/render-utils/src/ambient_occlusion.slf]: I know you just indented this code, and so the ridiculous formatting came from the previous author, but this really doesn't need to be so many lines...

**Pull request merged**